### DISCUSSION TITLE
register and edit MIME types and associations via NEMO

### DISCUSSION THREAD

**Discussion by AtomicRobotMan0101** [linuxmint/discussions#159] [Ideas]
**PROBLEM**

-- Nemo/cinnamon opens a file according to its type and content.
-- Example: If a file is an XML, it will be opened with the registered application for "plain text document"
-- Changing the type to open with with another application, it messes with all other text documents
-- Adding the _types-to-associations_ manually is not especially hard, but is a PITA (see last link, below)
-- The Nemo --> Properties --> Open With, does not cover this scenario

**SUGGESTION**

-- Create a tool, or modification to NEMO, which will allow the simple management of the extension-to-application associations.
-- Or place the tool in Cinnamemu, Preferences, System Settings app, as an icon (?)

**WHY?**

-- Example: using noMachine and a shortcut, which is placed on the desktop, the file is called foo.nxs. Its content is XML and opened as a text document. Changing the association to noMachine breaks all other non-associated documents with similar content.
-- Having a right-click, Open With, belies the purpose of having a desktop icon as a shortcut, for it saves no steps. Its expediency is lost.

**PREVIOUS MENTIONS**

A bit of a look around has shown this to be a minor issue, but for those coming from Windows-land, it is logical. As we move more towards the visual desktop paradigm and migrate Windows refugees, _they will expect and anticipate this behaviour._

See:
-- https://forums.linuxmint.com/search.php?st=0&sk=t&sd=d&sr=posts&keywords=register&fid%5B%5D=29
-- https://forums.linuxmint.com/viewtopic.php?p=1633235
-- (from 2009) https://rox.sourceforge.net/desktop/MIME-Editor
-- https://superuser.com/questions/459628/how-to-configure-linux-to-open-files-by-extension

**Comment by billyswong** [comment]
This sounds like a job for "Preferred Applications", if it is ever implemented in Linux Mint / Cinnamon.
